sql/catalyst/src/main/antlr3/org/apache/spark/sql/catalyst/parser/SparkSqlLexer.g

```diff
@@ -465,7 +467,7 @@ Identifier
 fragment    
 QuotedIdentifier 
     :
-    '`'  ( '``' | ~('`') )* '`' { setText(getText().substring(1, getText().length() -1 ).replaceAll("``", "`")); }
+    '`'  ( '``' | ~('`') )* '`' { setText(getText().replaceAll("``", "`")); }
```

Comment:
You are using the CatalystQl ```cleanIdentifier``` function to get rid of the remaining backticks. What happens if we do this:

    sql("select * from (select 1 as ```weird`name`) as a")